opal/mca/timer/linux/timer_linux_component.c

```diff
@@ -177,6 +178,20 @@ static int opal_timer_linux_find_freq(void)
         }
     }
 
+#if defined(PLATFORM_ARCH_RISCV)
+    if (0 == opal_timer_linux_freq) {
+        /* read timebase-frequency in device-tree */
+        FILE *fp_rv = fopen("/proc/device-tree/cpus/timebase-frequency", "rb");
+        if (NULL == fp_rv) {
+            return OPAL_ERR_IN_ERRNO;
+        }
+        if (1 == fread(&opal_timer_linux_freq, 4, 1, fp_rv)){
```

Comment:
I'm not familiar with the content of this file on risc64, but I wonder why it would store a value in a different endianness ? In other words do you really need to `ntohl` ?

Reply:
oh, I guess because of the  IEEE1275.